#! /usr/bin/env python
PACKAGE = "joystick2coms"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("CAN_ID_Decimal",    int_t,    0, "An Integer parameter", 103,16,102400)

hz_enum = gen.enum([gen.const("CAN_Stop",int_t,0,"CAN-BUS Stop"),
                      gen.const("CAN_Message_speed_1Hz",int_t,1,"CAN-BUS speed:1Hz"),
                      gen.const("CAN_Message_speed_10Hz",int_t, 2,"CAN-BUS speed:10Hz"),
                      gen.const("CAN_Message_speed_20Hz",int_t,3,"CAN-BUS speed:20Hz"),
                      gen.const("CAN_Message_speed_50Hz",int_t,4,"CAN-BUS speed:50Hz")],
                      "An enum to set CAN-BUS speed" )

gen.add("CAN_Speed_Hz",int_t,0,"The Speed of CAN-BUS",0,0,4,edit_method=hz_enum)

exit(gen.generate(PACKAGE,"joystick2coms","joystick"))
